tests/merge/analysis.h

```diff
@@ -0,0 +1,16 @@
+#ifndef INCLUDE_cl_merge_analysis_h__
+#define INCLUDE_cl_merge_analysis_h__
+
+void testimpl_merge_analysis__initialize(git_repository *t_repo, git_index *t_repo_index);
+void testimpl_merge_analysis__cleanup(void);
+void testimpl_merge_analysis__fastforward(void);
+void testimpl_merge_analysis__no_fastforward(void);
+void testimpl_merge_analysis__uptodate(void);
+void testimpl_merge_analysis__uptodate_merging_prev_commit(void);
+void testimpl_merge_analysis__unborn(void);
+void testimpl_merge_analysis__fastforward_with_config_noff(void);
+void testimpl_merge_analysis__no_fastforward_with_config_ffonly(void);
+void testimpl_merge_analysis__between_uptodate_refs(void);
+void testimpl_merge_analysis__between_noff_refs(void);
```

Comment:
I'm not really all that happy with this approach, but I guess it's the best we can do right now with what clar gives us. We probably should extend clar with some kind of logic that allows us to do data-driven tests like @tiennou proposes, ideally with not much change at all to how the rest of our tests  work. Meanwhile, this is fine with me.